code/game/objects/items/devices/spyglasses.dm

```diff
@@ -0,0 +1,91 @@
+//antag spyglasses. meant to be an example for map_popups.dm
+/obj/item/clothing/glasses/regular/spy
+	desc = "Made by Nerd. Co's infiltration and surveillance departmentt. Upon closer inspection, there's a small screen put into the lenses."
+	var/obj/item/spy_bug/linked_bug
+
+/obj/item/clothing/glasses/regular/spy/proc/show_to_user(var/mob/user)//this is the meat of it. most of the map_popup usage is in this.
+	if(!user)
+		return
+	if(!user.client)
+		return	
+	if(!linked_bug)
+		user.audible_message("<span class='warning'>[src] lets off a shrill beep!</span>")
+
+	if("spypopup_map" in user.client.screen_maps) //alright, the popup this object uses is already IN use, so the window is open. no point in doing any other work here, so we're good. 
+		return 
+	user.client.setup_popup("spypopup",3,3,2)
+	var/list/buglist = list(linked_bug.cam_view,linked_bug.popupmaster)
```

Comment:
how does this work? you're adding it to the screen loc of the popup window, but then it somehow sees the map view?

Reply:
`cam_view` is updated to include every turf in a range(1) in it's vis contents.